capstone-frontend/src/components/UserCard.jsx

```diff
@@ -7,36 +7,78 @@ import '../styles/Groups.css';
+      "add_memberIDs": this.props.user.id,
+      "id": this.props.club.id
+    };
+    fetch("/api/clubs", { method: "put", body: JSON.stringify(jsonBody) })
+      .then(this.props.fetchData)
+      .catch(function (err) {
+        //TODO #61: Centralize error output
+        alert(err);
+      });
+  }
+
+  rejectMember = () => {
+    const jsonBody = {
+      "remove_requestIDs": this.props.user.id,
+      "id": this.props.club.id
+    };
+    fetch("/api/clubs", { method: "put", body: JSON.stringify(jsonBody) })
+      .then(this.props.fetchData)
+      .catch(function (err) {
+        //TODO #61: Centralize error output
+        alert(err);
+      });
   }
 
   addFriend = () => {
     //TODO #86: Create friend functionality
   }
 
   render() {
-    const removeMember = this.props.club 
-                             && this.props.club.ownerID === window.localStorage.getItem("userID")
-                             && this.props.club.ownerID !== this.props.user.id
-                             && <Button id="remove-member" variant="danger" onClick={this.removeMember}>
-                                  Remove&nbsp;Member
-                                </Button>;
+    const isMember = this.props.club && this.props.club.memberIDs && this.props.club.memberIDs.includes(this.props.user.id);
+    const isRequester = this.props.club && !isMember && this.props.club.requestIDs && this.props.club.requestIDs.includes(this.props.user.id);
+    const removeMember = this.props.club
+      && this.props.club.ownerID === window.localStorage.getItem("userID")
+      && this.props.club.ownerID !== this.props.user.id
+      && <Button className="mt-2" cvariant="danger" onClick={this.removeMember}>
+        Remove&nbsp;Member
+         </Button>;
+    const requestButtons = this.props.club
+      && this.props.club.ownerID === window.localStorage.getItem("userID")
+      && this.props.club.ownerID !== this.props.user.id
+      && <div>
+        <Button className="mt-2" variant="primary" onClick={this.acceptMember}>
+          Accept&nbsp;Member
+           </Button>
+        <Button className="mt-2" variant="danger" onClick={this.rejectMember}>
+          Reject&nbsp;Member
+           </Button>
+      </div>
     return (
-      <Col className="user-card" xs={{span: "2"}} >
-        <img id="user-profile" src={this.props.user.profileImageUrl} alt="Profile"/>
-        <div> {this.props.user.fullName} </div> 
-        <Button variant="primary" onClick={this.addFriend}> Add Friend </Button>
-        {removeMember}
+      <Col className="user-card" xs={{ span: "2" }} >
+        <img id="user-profile" src={this.props.user.profileImageUrl} alt="Profile" />
+        <div> {this.props.user.fullName} </div>
+        <Button className="mt-2" variant="primary" onClick={this.addFriend}> Add Friend </Button>
+        {isMember && removeMember}
+        {isRequester && requestButtons}
       </Col>
     );
   }
-} 
- 
+}
+
 export default UserCard;
```

Comment:
You can turn this into a "named" component by putting export in front of the class declaration at the top!

Reply:
Will fix